### DISCUSSION TITLE
Multi selection ListView with ability to fast removing elements?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3622] [Q&A]
I'm searching for replacement for gtk 3/4 that I currently use in my apps.

Slint looks like good replacement for most components that I use, except one - listview, in which multiple elements can be selected and removed

This is how it works currently - I can easily select multiple items with mouse click + shift, then press space(this is default gtk behavior) and then remove random items from model.

https://github.com/slint-ui/slint/assets/41945903/fbb3fc71-cb54-4a76-ad68-43bcc3142a7c

It works fast even when removing thousands records from any place of listview without needing to replace model(not sure, but I only found solution with replacing models as solution to removing items -> remember model, remove item from it, override listview model)

Is it possible to do this now or later? Or maybe this feature is not planned?

**Comment by ogoffart** [comment]
> Is it possible to do this now or later?

It should be possible to do it now, but some things have to be done manually.
The model should contain a boolean whether the element is selected or not.  You can then use a ListView and in the element have a TouchArea that has a callback on click and forward that to the native code, including if shift is pressed. The native code can then select the range between the current item and the clicked item by changing the selection value in the model for all items

**Reply by qarmin** [reply]
And what about removing items from model?

Do I need to clone model,  remove items and then set new model as current?(A little wasteful)

**Reply by ogoffart** [reply]
You can use VecModel::remove.
Or if you write your own model, do the changes in your datastructure, and call ModelNotify::row_removed
